### PR TITLE
Integrated Cloud Firestore to store user data

### PR DESCRIPTION
**Pull request by weiyoung**
Signup popup now takes in first name, last name, email, and password
Screen Shot 2021-07-04 at 6 15 57 PM

First and last names are displayed after logging in
Screen Shot 2021-07-14 at 5 41 34 PM

user state now contains data from Cloud Firestore:
Screen Shot 2021-07-14 at 5 39 28 PM

### CODE REVIEW

File: src/contexts/useAuth.jsx [hunk 1]
```diff
@@ -21,9 +34,27 @@ const AuthProvider = ({children}) => {
...
+            plan: data.plan,
+            stripe_id: data.stripe_id,
+            watchlist: data.watchlist
+        })
+    }
+
     useEffect(() => {
         return auth.onAuthStateChanged(firebaseUser => {
-            setUser(firebaseUser)
+            if (firebaseUser != null) {
```
Comment (HGoyal09) [inline src/contexts/useAuth.jsx]: I am not sure why this is all under return

**Review by paulsgh — CHANGES_REQUESTED** [review]
see comment below

**Review by paulsgh — DISMISSED** [review]
minor:
since the approach we're using for the placeholder text is _example text_, then I think the `password confirmation` placeholder should be changed to "password", so it matches the `password` placeholder text. It should only be "password confirmation" if we were using the placeholder text as _labels_, e.g., "John" and "Doe" would instead be "First Name" and "Last Name"

File: src/contexts/useAuth.jsx (hunk 1 above)
Comment (weiyoung) [reply src/contexts/useAuth.jsx]: fixed in commit "fixed useEffect return, combined user name..."

File: src/components/LoginPopup/SignupForm.jsx [hunk 2]
```diff
@@ -18,9 +18,9 @@ const SignupForm = () => {
 
         try {
             setError("")
-            await signup(data.email, data.password)
-        } catch {
-            return setError("Failed to create account")
+            await signup(data.firstname + ' ' + data.lastname, data.email, data.password)
```
Comment (HGoyal09) [inline src/components/LoginPopup/SignupForm.jsx]: Lets change the function parameter to 
const onSumbit = async({password, passwordConfirmation, firstname, lastname, email}) => {
....
}

File: src/components/PageTemplate/Footer.jsx [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
-                  isExternal={true}
+                  isExternal
+                  p={0} bg="transparent" _hover={{color: "brand.500"}}>
+                <Icon as={FaGithub} w={6} h={6} m={MARGIN}/>
+            </Link>
+            <Link href="https://www.facebook.com/people/Investobull-Investors/100068350916511/"
+                  isExternal
                   p={0} bg="transparent" _hover={{color: "brand.500"}}>
-                <Icon as={FaGithub} w={6} h={6} m={2}/>
+                <Icon as={FaFacebookSquare} w={6} h={6} m={MARGIN}/>
```
Comment (HGoyal09) [inline src/components/PageTemplate/Footer.jsx]: Code repitition 
Would be a nice idea to declare a custom Icon on the top as a component and specify the common fields - w, h, m there and send as an as other prop

File: src/components/PageTemplate/Navbar.jsx [hunk 4]
```diff
@@ -74,7 +75,7 @@ const Links = () => (
         <Button bg="transparent" m={MARGIN} as={Link} to="/">Home</Button>
         <Button bg="transparent" m={MARGIN} as={Link} to="/watchlist">Watchlist</Button>
         <Button bg="transparent" m={MARGIN} as={Link} to="/news">News</Button>
-        <Button bg="transparent" m={MARGIN} as={Link} to="/about">About</Button>
+        <Button bg="transparent" m={MARGIN} as={Link} to="/plans">Plans</Button>
```
Comment (HGoyal09) [inline src/components/PageTemplate/Navbar.jsx]: Would be a better idea to not repeat css configs and have a customNavbarButton as a component

**Review by HGoyal09 — APPROVED** [review]

File: src/components/LoginPopup/SignupForm.jsx (hunk 2 above)
Comment (weiyoung) [reply src/components/LoginPopup/SignupForm.jsx:21]: sorry, why?
we don't need the password confirmation in the function, we can check it in the frontend.
I combined the firstname and lastname here so it's cleaner later on.

Comment (weiyoung) [reply src/components/LoginPopup/SignupForm.jsx]: wait, you commented on the wrong line, I thought you were referring to something else.
not a big difference but sure I can do that 👌🏼

File: src/components/PageTemplate/Footer.jsx (hunk 3 above)
Comment (weiyoung) [reply src/components/PageTemplate/Footer.jsx]: will fix 👌🏼

**Review by HGoyal09 — APPROVED** [review]

**Review by paulsgh — APPROVED** [review]

**Pull request merged**